### PR TITLE
CVE fixes of High priority

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
# Issue link

Jira: https://issues.redhat.com/browse/RHOAIENG-5504 

# What changes have been made

Upgraded dependencies to fix a number of High CVEs.

# Verification steps

e2e tests passing is sufficient.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,6 +22,10 @@ require (
 
 replace sigs.k8s.io/custom-metrics-apiserver => sigs.k8s.io/custom-metrics-apiserver v1.25.1-0.20230306170449-63d8c93851f3
 
+replace go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp => go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.44.0
+
+replace github.com/jackc/pgx/v4 => github.com/jackc/pgx/v5 v5.5.4
```
Comment (KPostOffice) [inline go.mod:27]: Can this be change to v5.5.5?

**Review by KPostOffice — COMMENTED** [review]
One tiny comment

File: go.mod (hunk 1 above)
Comment (ChristianZaccaria) [reply go.mod:27]: It is a strange one, I tried doing that but `go` doesn't like it. I.e.,

- With v5.5.4, a `go mod tidy` successfully replaces the v4 module in `go.mod` and `go.sum`.
- With v5.5.5 a `go mod tidy` fails.

I even performed a `git reset --hard` to previous commit to ensure a fresh state to test getting and replacing the module for v5.5.5, but still failed on `go mod tidy`.

```
[christianzaccaria@thinkpad codeflare-operator]$ go get github.com/jackc/pgx/v5@v5.5.4
go: github.com/jackc/pgx/v5@v5.5.4 used for two different module paths (github.com/jackc/pgx/v4 and github.com/jackc/pgx/v5)
[christianzaccaria@thinkpad codeflare-operator]$ go mod tidy
[christianzaccaria@thinkpad codeflare-operator]$ go get github.com/jackc/pgx/v5@v5.5.5
go: github.com/jackc/pgx/v5@v5.5.5 used for two different module paths (github.com/jackc/pgx/v4 and github.com/jackc/pgx/v5)
[christianzaccaria@thinkpad codeflare-operator]$ go mod tidy
go: finding module for package github.com/jackc/pgx/v5/pgxpool
go: finding module for package github.com/jackc/pgx/v5/pgconn
go: finding module for package github.com/jackc/pgx/v5/pgtype
go: finding module for package github.com/jackc/pgx/v5
go: found github.com/jackc/pgx/v5 in github.com/jackc/pgx/v5 v5.5.5
go: found github.com/jackc/pgx/v5/pgconn in github.com/jackc/pgx/v5 v5.5.5
go: found github.com/jackc/pgx/v5/pgtype in github.com/jackc/pgx/v5 v5.5.5
go: found github.com/jackc/pgx/v5/pgxpool in github.com/jackc/pgx/v5 v5.5.5
go: github.com/jackc/pgx/v5@v5.5.5 used for two different module paths (github.com/jackc/pgx/v4 and github.com/jackc/pgx/v5)
```

**Review by astefanutti — COMMENTED** [review]
Out-of-curiosity, where do these transitive dependencies come from? Could they be upgraded by upgrading k8s version / controller-runtime?

**Comment by ChristianZaccaria** [comment]
> Out-of-curiosity, where do these transitive dependencies come from? Could they be upgraded by upgrading k8s version / controller-runtime?

@astefanutti For `pgx` module, the transitive dependencies come from `ocm-sdk-go`. Their main branch and also latest tag make use of an older version of `pgx` module.

The other module `otelhttp` does come from `k8s.io/component-base` and `k8s.io/apiextensions-apiserver`. I'll see if I can upgrade these ones instead, thanks!

**Comment by sutaakar** [comment]
The test issues could be resolved by upgrading `codeflare-common` , however upgraded version use Go 1.21, so it can't be used until we upgrade CFO to 1.21?

**Comment by ChristianZaccaria** [comment]
@sutaakar Thank you, that could work. I see that the latest go ubi8 image is `1.20`. Do you think dropping the latest commit here for now is okay?

**Comment by sutaakar** [comment]
it is fine for me

**Comment by Bobbins228** [comment]
When trying to create an image I got this error:
```
/home/mark/codeflare-operator/bin/controller-gen rbac:roleName=manager-role webhook paths="./..."
Error: err: exit status 1: stderr: go: updates to go.mod needed; to update it:
	go mod tidy
run `controller-gen rbac:roleName=manager-role webhook paths=./... -w` to see all available markers, or `controller-gen rbac:roleName=manager-role webhook paths=./... -h` for usage
make: *** [Makefile:131: manifests] Error 1
```

**Comment by Fiona-Waters** [comment]
I tried to create an image with `make image-build -e IMG=quay.io/rh_ee_fwaters/codeflare-operator:cve` and it worked successfully. I am running go1.20.
@Bobbins228 is using go1.22.1 so this could be what is causing the issue.

**Comment by astefanutti** [comment]
@ChristianZaccaria could you give this a rebase please?

**Comment by ChristianZaccaria** [comment]
> @ChristianZaccaria could you give this a rebase please?

Rebased, thank you!

**Pull request merged**